jetty-server/src/test/java/org/eclipse/jetty/server/GracefulStopTest.java

```diff
@@ -657,6 +659,41 @@ public void handle(String target, Request baseRequest, HttpServletRequest reques
         assertThat(response,Matchers.not(Matchers.containsString("Connection: close")));
         assertTrue(latch.await(10,TimeUnit.SECONDS));
     }
+
+    @Test
+    public void testFailedStart()
+    {
+        Server server= new Server();
+
+        LocalConnector connector = new LocalConnector(server);
+        server.addConnector(connector);
+
+        ContextHandlerCollection contexts = new ContextHandlerCollection();
+        server.setHandler(contexts);
+        ContextHandler context0 = new ContextHandler(server,"/zero");
+        ContextHandler context1 = new ContextHandler(server,"/one")
+        {
+            @Override
+            protected void doStart() throws Exception
+            {
+                throw new Exception("Test start failure");
+            }
+        };
+        contexts.setHandlers(new Handler[]{context0,context1});
+
+        try
+        {
+            server.start();
+            fail();
+        }
+        catch(Exception e)
+        {
+            assertThat(e.getMessage(),is("Test start failure"));
+        }
+
+        assertTrue(server.getContainedBeans(LifeCycle.class).stream().noneMatch(LifeCycle::isRunning));
+        assertTrue(server.getContainedBeans(LifeCycle.class).stream().anyMatch(LifeCycle::isFailed));
+    }
```

Comment:
Can you please add a test case that proves that a third context is not started if the second throws? And that is also not tried to be stopped?
This new test will cement the purpose of this PR and the reason why we don't call `stop()` in case of failures, but rather we catch and iterate and only stop the running beans.

Reply:
@gregw you're not testing that `context2.stop()` has _not_ been called.